pool/pool_aio_test.go

```diff
@@ -105,28 +145,127 @@ func TestPoolInterfaceWithAIO(t *testing.T) {
+		cl, _, err := pool.sdkClient()
 
-		containerID, err = pool.ContainerPut(ctx, cont, cmd)
 		require.NoError(t, err)
+		require.NoError(t, isEACLCreated(ctxTimeout, cl, containerID, table))
+	})
+
+	t.Run("upload object", func(t *testing.T) {
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
+		defer cancel()
 
-		ctxTimeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		objectID = testObjectPutInit(t, ctxTimeout, account, containerID, signer, payload, pool)
+	})
+
+	t.Run("download object", func(t *testing.T) {
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
 		defer cancel()
 
-		err = isBucketCreated(ctxTimeout, pool, containerID)
+		var cmd client.PrmObjectGet
+		cmd.UseSigner(signer)
+
+		hdr, read, err := pool.ObjectGetInit(ctxTimeout, containerID, objectID, cmd)
+		defer func() {
+			_ = read.Close()
+		}()
+
 		require.NoError(t, err)
+		require.NotNil(t, hdr.OwnerID())
+		require.True(t, hdr.OwnerID().Equals(account))
+
+		downloadedPayload := make([]byte, len(payload))
+
+		l, ok := read.ReadChunk(downloadedPayload)
+		require.True(t, ok)
+		require.Equal(t, l, len(payload))
+
+		require.True(t, bytes.Equal(payload, downloadedPayload))
+	})
+
+	t.Run("delete object", func(t *testing.T) {
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
+		defer cancel()
+
+		testDeleteObject(t, ctxTimeout, signer, containerID, objectID, pool)
+		cl, _, err := pool.sdkClient()
+
+		require.NoError(t, err)
+		require.NoError(t, isObjectDeleted(ctxTimeout, cl, containerID, objectID))
+	})
+
+	t.Run("container delete", func(t *testing.T) {
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
+		defer cancel()
+
+		testDeleteContainer(t, ctxTimeout, signer, containerID, pool)
+		cl, _, err := pool.sdkClient()
+
+		require.NoError(t, err)
+		require.NoError(t, isBucketDeleted(ctxTimeout, cl, containerID))
+	})
+}
+
+func TestPoolWaiterWithAIO(t *testing.T) {
+	ctx := context.Background()
+
+	account, signer, cont := testData(t)
+
+	opts := InitParameters{
+		signer: signer,
+		nodeParams: []NodeParam{
+			{1, "grpc://localhost:8080", 1},
+		},
+		clientRebalanceInterval: 30 * time.Second,
+	}
+
+	pool, err := NewPool(opts)
+	require.NoError(t, err)
+	require.NoError(t, pool.Dial(ctx))
+
+	var containerID cid.ID
+	var objectID oid.ID
+
+	payload := make([]byte, 8)
+	_, err = rand.Read(payload)
+	require.NoError(t, err)
+
+	defaultPoolingTimeout := 1 * time.Second
+	wait := waiter.NewWaiter(pool, defaultPoolingTimeout)
+
+	t.Run("create container", func(t *testing.T) {
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
+		defer cancel()
+
+		containerID = testCreateContainer(t, ctxTimeout, signer, cont, wait)
+	})
+
+	t.Run("set eacl", func(t *testing.T) {
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
+		defer cancel()
+
+		testSetEacl(t, ctxTimeout, signer, containerID, wait)
```

Comment:
If we have waited for the result then we should check for the appropriate side-effect immediately after the `test..` execution.

Reply:
Added one more test, to check the eacl Table created properly